pyrokinetics/gk_code/GKOutputReaderGS2.py

```diff
@@ -166,36 +188,57 @@ def _set_fields(
         if not np.any(np.isin(field_names, raw_data.data_vars)):
             return data
 
-        coords = ["field", "theta", "kx", "ky", "time"]
-        fields = np.empty([data.dims[coord] for coord in coords], dtype=complex)
+        coords = ["theta", "kx", "ky", "time"]
+
+        local_norm = data.local_norm
+        pyro_field_units = field_units(local_norm.pyrokinetics)
+        gs2_field_units = field_units(local_norm.gs2)
 
         # Loop through all fields and add field if it exists
-        for ifield, field_name in enumerate(field_names):
+        for ifield, (field_name, field_unit) in enumerate(
+            zip(field_names, gs2_field_units.values())
+        ):
+            field = np.empty([data.dims[coord] for coord in coords], dtype=complex)
+
             if field_name not in raw_data:
                 logging.warning(
                     f"Field data over time {field_name} not written to netCDF file. "
                     "Setting this field to 0"
                 )
                 # Note: we could instead set this to the field at the last time step
-                fields[ifield, :, :, :, :] = 0
+                field[:, :, :, :] = 0
                 continue
 
             # raw_field has coords (t,ky,kx,theta,real/imag).
             # We wish to transpose that to (real/imag,theta,kx,ky,t)
             field_data = raw_data[field_name].transpose("ri", "theta", "kx", "ky", "t")
-            fields[ifield, :, :, :, :] = sqrt2 * (
+            field = (
                 field_data[0, :, :, :, :] + 1j * field_data[1, :, :, :, :]
```

Comment:
Just to note that numpy lets you do `...` for "the rest of the indices":

```suggestion
                field_data[0, ...] + 1j * field_data[1, ...]
```

Reply:
First time I've seen this used in a real case. [Relevant stack overflow ](https://stackoverflow.com/questions/772124/what-does-the-ellipsis-object-do) for one of the most obscure bits of Python syntax.

Reply:
Ooooh that's good to know!